### PR TITLE
Add support for handling Target and TargetEmails (WIP)

### PR DESCRIPTION
**Pull request by marcinkoziej**

### CODE REVIEW

File: proca/lib/proca/target.ex [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  alias Proca.{Repo, Target}
+  import Ecto.Changeset
+  import Ecto.Query
+
+  @primary_key {:id, Ecto.UUID, autogenerate: true}
+
+  schema "targets" do
+    field :name, :string
+    field :external_id, :string
+    field :area, :string
```
Comment (tttp) [inline proca/lib/proca/target.ex:16]: what's the default? (double checking it can be null)

File: proca/lib/proca_web/schema/target_types.ex [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  input_object :target_email do
+    field :email, non_null(:string)
+  end
+
+  input_object :target_input do
+    field :name, non_null(:string)
+    field :area, :string
+    field :campaign_id, non_null(:integer)
+    field :external_id, non_null(:string)
+    field :fields, :json
```
Comment (marcinkoziej) [inline proca/lib/proca_web/schema/target_types.ex:19]: Lets make non-null and default to {}

File: proca/lib/proca_web/schema/target_types.ex [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  object :target do
+    field :id, non_null(:string)
+    field :name, non_null(:string)
+    field :area, :string
+    field :fields, :json
+  end
+
+  object :target_mutations do
+    field :upsert_targets, type: non_null(list_of(:target)) do
```
Comment (marcinkoziej) [inline proca/lib/proca_web/schema/target_types.ex:31]: Can we add campaign_id as to argument. so it's not repeated in every target

File: proca/lib/proca/target_email.ex [hunk 4]
```diff
@@ -0,0 +1,22 @@
+defmodule Proca.TargetEmail do
+  @moduledoc """
+  TargetEmail contains the email data for a Target
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "target_emails" do
```
Comment (tttp) [inline proca/lib/proca/target_email.ex:9]: I don't get splitting target and target email, the use cases I can imaging having something different than a 1-1:
- keeping a log (but you'd need a status active/inactive)
- have alternate emails (needs status/role too)

is this possible to have more than one email per target?

File: proca/lib/proca_web/schema/target_types.ex [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      arg :targets, non_null(list_of(:target_input))
+
+      resolve(&Resolvers.Target.upsert_targets/3)
+    end
+  end
+
+  object :target_queries do
+    field :targets, non_null(list_of(:target)) do
```
Comment (marcinkoziej) [inline proca/lib/proca_web/schema/target_types.ex:39]: can the targets query be a field of private_campaign and public_campaign ( but the targets under private_campaign should show all fields of target, and targets under private_campaign just a public subset).

File: proca/lib/proca_web/schema/eci_schema.ex [hunk 6]
```diff
@@ -17,6 +17,7 @@ defmodule ProcaWeb.Schema.EciSchema do
   import_types(ProcaWeb.Schema.OrgTypes)
   import_types(ProcaWeb.Schema.UserTypes)
   import_types(ProcaWeb.Schema.SubscriptionTypes)
+  import_types(ProcaWeb.Schema.TargetTypes)
```
Comment (tttp) [inline proca/lib/proca_web/schema/eci_schema.ex:20]: eci shouldn't be changed, they will never have targets

File: proca/schema.graphql [hunk 7]
```diff
@@ -64,6 +66,8 @@ type PublicCampaign implements Campaign {
 
   org: Org!
 
+  targets: [Target]
```
Comment (tttp) [inline proca/schema.graphql:69]: I don't see the use case here. @marcinkoziej do you?

**Review by tttp — CHANGES_REQUESTED** [review]

File: proca/lib/proca_web/schema/target_types.ex (hunk 3 above)
Comment (hiemanshu) [reply proca/lib/proca_web/schema/target_types.ex:31]: This allows to maintain 1 airtable per org, and have multiple targets in the same sheet with the campaign id as a parameter. It's what we said we'd start from during the call.

File: proca/lib/proca_web/schema/target_types.ex (hunk 5 above)
Comment (hiemanshu) [reply proca/lib/proca_web/schema/target_types.ex:39]: So the private campaign allows showing emails as well?

File: proca/lib/proca/target.ex (hunk 1 above)
Comment (hiemanshu) [reply proca/lib/proca/target.ex:16]: name, external_id and emails are required. Area and Fields can be null

File: proca/lib/proca_web/schema/eci_schema.ex (hunk 6 above)
Comment (hiemanshu) [reply proca/lib/proca_web/schema/eci_schema.ex:20]: The types wouldn't load without this. Marcin said it was required to import it here.

File: proca/schema.graphql (hunk 7 above)
Comment (marcinkoziej) [reply proca/schema.graphql]: Well to lost the campaign targets if you do not have cached it in static file

File: proca/lib/proca_web/schema/eci_schema.ex (hunk 6 above)
Comment (marcinkoziej) [reply proca/lib/proca_web/schema/eci_schema.ex:20]: Yes eci does not use this type in schema but absinthe will report missing type

File: proca/lib/proca/target_email.ex (hunk 4 above)
Comment (marcinkoziej) [reply proca/lib/proca/target_email.ex:9]: We talked on signal

File: proca/lib/proca_web/schema/target_types.ex (hunk 5 above)
Comment (marcinkoziej) [reply proca/lib/proca_web/schema/target_types.ex:44]: Yeah private is for dashboard for example, if we would like to list the emails somewhere

File: proca/lib/proca_web/schema/target_types.ex (hunk 3 above)
Comment (marcinkoziej) [reply proca/lib/proca_web/schema/target_types.ex:36]: Hmmm wouldn't having one airtable per campaign be better ? It's more self contained and campaigners cannot break one campaign by editing targets to another.
I remember we talked about being ok with duplication of targets (so for new campaign they will just clone the airtable, and there will be duplication but this is fine)

Comment (hiemanshu) [reply proca/lib/proca_web/schema/target_types.ex:35]: Changed

**Pull request merged**